apps/srp.c

```diff
@@ -553,6 +553,8 @@ int srp_main(int argc, char **argv)
             user = *(argv++);
         else {
             user = NULL;
+            if (mode == OPT_LIST)
+                break;
```

Comment:
Added two commits addressing the issues raised above.
As I am new to committing code to OpenSSL I took the "shy" approach and did minimal changes under the principle that if it was done one way it was for a good reason, but I'm getting more confident :-)